lib/Secrets/interactionrequest.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "Secrets/interactionrequest.h"
+#include "Secrets/interactionrequest_p.h"
+
+#include "Secrets/secretmanager.h"
+#include "Secrets/secretmanager_p.h"
+#include "Secrets/serialisation_p.h"
+
+#include <QtDBus/QDBusPendingReply>
+#include <QtDBus/QDBusPendingCallWatcher>
+
+using namespace Sailfish::Secrets;
+
+InteractionRequestPrivate::InteractionRequestPrivate()
+    : m_status(Request::Inactive)
+{
+}
+
+/*!
+ * \class InteractionRequest
+ * \brief Allows a client request user input, mediated by the secrets service
+ *
+ * This class allows clients to request user input, via a system-mediated
+ * user interaction flow.  The user will be explicitly informed that the
+ * application has initiated the user interaction request, and that the
+ * supplied data should not be considered to be secure.
+ *
+ * Most applications should not have any need to use this request type,
+ * as they can usually show their own input dialog and ask the user for
+ * input directly that way, however some daemon services without any
+ * UI capability may need to use this request type to retrieve
+ * non-sensitive or transient data from the user.
+ * *
+ * An example of retrieving user input follows:
+ *
+ * \code
+ * // Define the interaction request prompt parameters.
+ * Sailfish::Secrets::InteractionParameters uiParams;
+ * uiParams.setInputType(Sailfish::Secrets::InteractionParameters::AlphaNumericInput);
+ * uiParams.setEchoMode(Sailfish::Secrets::InteractionParameters::NormalEcho);
+ * uiParams.setPromptText(tr("Enter some data"));
+ *
+ * // Perform the request.
+ * Sailfish::Secrets::SecretManager sm;
+ * Sailfish::Secrets::InteractionRequest ir;
+ * ir.setManager(&sm);
+ * ir.startRequest(); // status() will change to Finished when complete
+ * ir.waitForFinished(); // or better: connect to statusChanged()
+ * QByteArray userInput = ir.userInput();
+ * \endcode
+ */
+
+/*!
+ * \brief Constructs a new InteractionRequest object with the given \a parent.
+ */
+InteractionRequest::InteractionRequest(QObject *parent)
+    : Request(parent)
+    , d_ptr(new InteractionRequestPrivate)
+{
+}
+
+/*!
+ * \brief Destroys the InteractionRequest
+ */
+InteractionRequest::~InteractionRequest()
+{
+}
+
+/*!
+ * \brief Returns the interaction parameters which allow customisation of the input prompt
+ */
+InteractionParameters InteractionRequest::interactionParameters() const
+{
+    Q_D(const InteractionRequest);
+    return d->m_interactionParameters;
+}
+
+/*!
+ * \brief Sets the interaction parameters which allow customisation of the input prompt to \a uiParams
+ */
+void InteractionRequest::setInteractionParameters(const InteractionParameters &uiParams)
```

Comment:
Why is it called set InteractionParameters() while it is called setUiParameters() in storesecretrequest?

Reply:
Good point.  We should standardise.  I guess we should use setInteractionParameters() everywhere for consistency.  I will update those other cases.